components/blitz/src/omero/gateway/facility/ROIFacility.java

```diff
@@ -490,29 +490,24 @@ public ROIResult loadROI(SecurityContext ctx, long roiId)
                                     .copyFolderLinks()) {
                                 if (link.getParent().getId().getValue() == folder
                                         .getId()) {
-                                    updateService.deleteObject(link);
-                                    
-                                    ROIData rd = new ROIData(link.getChild());
-                                    FolderData fd = new FolderData(
-                                            link.getParent());
-
-                                    if (getById(result, fd) == null) {
-                                        result.put(fd, new ArrayList<ROIData>());
-                                    }
-                                    getById(result, fd).add(rd);
+                                    toDelete.add(link.getId().getValue());
                                 }
                             }
                         }
                     }
                 }
             }
-            
-            return result;
-        } catch (Exception e) {
-            handleException(this, e, "Cannot add ROIs to Folder ");
-            return Collections.EMPTY_MAP;
-        }
 
+            if (!toDelete.isEmpty()) {
+                Request req = Requests.delete().target("FolderRoiLink")
+                        .id(toDelete).build();
```

Comment:
I was wondering about that, too. Just tried without childoption and ROIs were still there :)